src/Agoda.Analyzers/AgodaCustom/CustomRulesResources.resx

```diff
@@ -156,4 +156,7 @@
   <data name="AG0005Title" xml:space="preserve">
     <value>Test method names must follow convention</value>
   </data>
+  <data name="AG0006Title" xml:space="preserve">
+    <value>You should not use expensive reflection methods.</value>
```

Comment:
The purpose of this rule shouldn't be to prevent "expensive" reflection methods, but to prevent using strings to identify types for activation, which is not friendly for refactoring or compile time checking.